### PR TITLE
Poolseq

### PR DESCRIPTION
**Pull request by robmaz**
An attempt at an installer script for poolSeq. I am curious what you think about that.

### CODE REVIEW

File: Formula/poolseq.rb [hunk 1]
```diff
@@ -0,0 +1,40 @@
+class Poolseq < Formula
+  desc "Analyze and simulate Pool-Seq time series data"
+  homepage "https://github.com/ThomasTaus/poolSeq/"
+  url "https://github.com/ThomasTaus/poolSeq/releases/download/v0.3.1/poolSeq_0.3.1.tar.gz"
+  sha256 "a38a1a0d79ec16ba6ec9227d3f61f037154146e1b882940ca013f8f98db43eb3"
+
+  bottle :unneeded
+
+  # R is not really optional, but may not be managed by Homebrew
```
Comment (magicDGS) [inline Formula/poolseq.rb]: This should be added to caveats too...

File: Formula/poolseq.rb [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+    system "Rscript", "install-poolSeq.r"
+    $CHILD_STATUS.exitstatus.nil &&
+      raise("R may be missing or not in your PATH. To install R, use the --with-r option.")
+  end
+
+  def caveats
+    <<-EOS.undent
+      This installer just called R's install.packages() function and did not
+      install any files itself. "brew uninstall" will only remove the entry in
+      Homebrew's list of installed packages, but not the actual R package.
```
Comment (magicDGS) [inline Formula/poolseq.rb]: Maybe you can define a uninstall function in the formula and call [`remove.packages()`](https://stat.ethz.ch/R-manual/R-devel/library/utils/html/remove.packages.html) __only__ for _poolSeq_. I think that it is necessary for a formula to being able to really uninstall the package itself (not the dependencies).

**Comment by robmaz** [comment]
It does not quite work anyway.The R script works on the command line,
but I think it may execute in some sandbox or something, because there
are no packages installed and the test fails. I need to rethink this,
probably
find the R lib directory and install the files there properly. Then they
could
also be uninstalled.

Thanks for looking
Cheers
Rupert

2017-10-10 15:05 GMT+02:00 Daniel Gómez-Sánchez <notifications@github.com>:

**Comment by robmaz** [comment]
Just committed a version that works locally. The downside is, it requires a Homebrew-managed R installation. I don't know how to find and install into a valid R library folder otherwise.

**Pull request merged**